### ISSUE TITLE
-Werror will break oneAPI build

### ISSUE THREAD

**Issue by aminiussi** [boostorg/mysql#173]
OneAPI build easily end up with unsupported options inherited from legacy Intel. These are warning with oneAPI, maklng them errors will break the build.

The following patch seems to fix the issue:
```
12:44:33 [alainm@castor mysql]# git diff -c
diff --git a/cmake/test_utils.cmake b/cmake/test_utils.cmake
index 2bf0cc7..c77cbd2 100644
--- a/cmake/test_utils.cmake
+++ b/cmake/test_utils.cmake
@@ -38,6 +38,8 @@ function(boost_mysql_common_target_settings TARGET_NAME)
             _SILENCE_CXX17_ADAPTOR_TYPEDEFS_DEPRECATION_WARNING # Warnings in C++17 for Asio
         )
         target_compile_options(${TARGET_NAME} PUBLIC /bigobj) # Prevent failures on Windows
+    elseif (CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
+        target_compile_options(${TARGET_NAME} PUBLIC -Wall -Wextra -pedantic)
     else()
         target_compile_options(${TARGET_NAME} PUBLIC -Wall -Wextra -pedantic -Werror)
     endif()
12:44:37 [alainm@castor mysql]# 
```

**Comment by anarthal** [comment]
Sorry for the late answer. I've reproduced the warnings, which point to issues with Intel's compiler handling of floating-point numbers. For example:

```
/home/ruben/workspace/boost-root/libs/mysql/include/boost/mysql/impl/internal/protocol/deserialize_binary_field.ipp:104:9: error: explicit comparison with NaN in fast floating point mode [-Werror,-Wtautological-constant-compare]
    if (std::isnan(v) || std::isinf(v))
```

Can you please confirm that these are the warnings you're seeing?

Just disabling them looks like a bad idea, since these checks are there for a reason - these values represent database values that can be controlled by a user, which has security implications.

If you're using cmake to test Boost.MPI in test mode, I strongly recommend just building MPI by using

```
cmake -DCMAKE_BUILD_TYPE=Release -DBOOST_INCLUDE_LIBRARIES=mpi -DBOOST_ENABLE_MPI=ON -DBUILD_TESTING=ON ../
```

Otherwise, you're building and running all the tests for all Boost libraries, which must be incredibly expensive.

I'll try to set up a CI for the Intel compiler and fix these warnings for 1.84, in case anyone needs to build MySQL under oneAPI - I've raised #174 for it.

Please let me know if `-DBOOST_INCLUDE_LIBRARIES=mpi` works for you, and I'll close this issue.

Regards,
Ruben.

**Comment by anarthal** [comment]
I'm assuming this worked.
